homeassistant/components/knx/weather.py

```diff
@@ -124,3 +135,26 @@ def wind_speed(self) -> float | None:
             if self._device.wind_speed is not None
             else None
         )
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
+        """Return device specific state attributes."""
+        attr: dict[str, Any] = {}
+        # pylint: disable=protected-access
+        if self._device._brightness_north.initialized:
+            attr[ATTR_BRIGHTNESS_NORTH] = self._device.brightness_north
```

Comment:
I wonder, should these be separate attributes or entities?

I think it would be really beneficial for cases like these, to implement the notion of Devices with this integration.

For that reason, I'm kinda in doubt if we should merge this or not.

Reply:
No one is stopping a user from configuring a sensor with the same group address used in a weather entity.
We use the brightness configured in the weather entity to calculate the current condition, so why not also show its value...

I'm not sure how the device abstraction would work here. Would the device hold a weather entity and some sensor entities? If so it would just save some lines of yaml?!

Reply:
Well, as a general rule of thumb, if Ine needs to automate on an entity attribute, it should have been a separate entity.

Reply:
Im ok with that. It allows for finer control over the sensor entity anyway. 
It was user request "WTH, I configured brightness GAs but don't see the values..."

Reply:
Discussed this with a couple of people. We agree that these attributes should not be added.
We don't think we should add extra attributes to weather platforms/entities at all. The weather entity is mainly for frontend display purposes, additional attributes will not add anything to that purpose.

Instead, separate sensor entities should be used.